### PR TITLE
Inline markdown images

### PR DESCRIPTION
**Pull request by tiagoalvesdulce**
Closes https://github.com/decred/politeiagui/issues/2143

### Solution description

This diff allows the user to reference images in the markdown. In order to do that I had to bump `react-mde` package to `10.3.0-alpha1` and I created the new `customSaveImageCommand`.

#### Use Cases

You should be able to copy and paste and to drag and drop images in the markdown editor. Also, there's a new `image` icon that can be used to include a new image.

Screen_Shot_2020-10-13_at_16_38_47

### UI Changes Screenshot

localhost_3000_ (1)

**TODO**

- [x] Do not show the images referenced in the markdown in the image gallery
- [ ] ~Fix proposal diff~
- [x] Update image attach modal copy to remove .txt
- [x] Don't allow users to attach .txt

The way we do proposals DIff right now is really difficult to render images in place with a good performance.

### CODE REVIEW

**Review by victorgcramos — COMMENTED** [review]
Hey @tiagoalvesdulce thanks for the PR. Code is looking really good as always!

One thing I noticed is that MIME types validation should be included in order to avoid image submission errors. Besides that, It's all good for this first version.

**Comment by degeri** [comment]
we had alot of security issues due to rendering images thats why we chose to not allow it during launch. Allowing someone to let the user load content is a risk.

eg:

https://github.com/decred/politeiagui/issues/681

**Comment by tiagoalvesdulce** [comment]
I fixed the proposal editing route, improved the error message for images with mime types not allowed (like `svg`) and prevent usage of images in the comments. I'll be completing the itens of the todo list in the description. Let me know if I need to add something else.

**Comment by degeri** [comment]
images in comments is still allowed its just not showing the option. We need to completely disable images from being rendered in comments. 

image

**Comment by tiagoalvesdulce** [comment]
Sorry, I forgot to change from image renderer to link renderer on comments. Should be good now. Only need to polish the Diff and the PR will be ready.

**Comment by degeri** [comment]
Checked. Looks good.

**Review by victorgcramos — APPROVED** [review]
LGTM!

File: src/components/Markdown/helpers.js [hunk 1]
```diff
@@ -80,30 +62,51 @@ const LinkRenderer = ({ url, children }) => {
   );
 };
 
-const imageHandler = ({ src, alt }) => {
-  return <LinkRenderer url={src}>{alt}</LinkRenderer>;
+// Use external link renderer when images are not allowed
+const imageHandler = (renderImages) => ({ src, alt }) => {
+  return renderImages ? (
```
Comment (amassarwi) [inline src/components/Markdown/helpers.js:67]: No need for the return here

File: src/components/MarkdownEditor/commands.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  image: imageSVG
+};
+
+/**
+ * Returns the React node to be displayed for a given command name.
+ * @param {string} commandName
+ * @returns {Object} React node
+ */
+export const getCommandIcon = (commandName) => {
+  return <img alt={commandName} src={commandsIcons[commandName]} />;
```
Comment (amassarwi) [inline src/components/MarkdownEditor/commands.js:60]: Ditch return

File: src/components/Proposal/Proposal.jsx [hunk 3]
```diff
@@ -144,6 +172,11 @@ const Proposal = React.memo(function Proposal({
   const { themeName } = useTheme();
   const isDarkTheme = themeName === DEFAULT_DARK_THEME_NAME;
 
+  const { text, markdownFiles } = replaceImgDigestWithPayload(
```
Comment (amassarwi) [inline src/components/Proposal/Proposal.jsx:175]: Could we use useMemo here ?

File: src/components/ProposalForm/ProposalForm.jsx [hunk 4]
```diff
@@ -286,9 +360,14 @@ const ProposalFormWrapper = ({
             );
           }
         }
+        const { description, files } = replaceBlobsByDigestsAndGetFiles(
```
Comment (amassarwi) [inline src/components/ProposalForm/ProposalForm.jsx:363]: Usememo ?

**Review by amassarwi — APPROVED** [review]
Nice work, added few minor suggestions 👍

File: src/components/ProposalForm/ProposalForm.jsx (hunk 4 above)
Comment (tiagoalvesdulce) [reply src/components/ProposalForm/ProposalForm.jsx:363]: This is inside an useCallback so I can't use useMemo

Comment (amassarwi) [reply src/components/ProposalForm/ProposalForm.jsx:363]: opps

**Pull request merged**